src/core/buffers/representation/representation_buffer.ts

```diff
@@ -265,9 +265,16 @@ export default function RepresentationBuffer<T>({
                            ...neededSegments ];
       }
 
+      const lastPosition = representation.index.getLastPosition();
+      const endOfPeriodIsAvailable = lastPosition != null &&
+                                     period.end != null &&
+                                     lastPosition >= period.end;
```

Comment:
Maybe less risky to implement as a method from RepresentationIndex with a tailored answer from each.

Also, should always return true for static contents.